pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -174,15 +176,18 @@ def vip_reparam_node(
 ) -> Tuple[ModelDeterministic, ModelNamed]:
     if not isinstance(node.op, RandomVariable | SymbolicRandomVariable):
         raise TypeError("Op should be RandomVariable type")
-    rv = node.default_output()
-    try:
-        [rv_shape] = constant_fold([rv.shape])
-    except NotConstantValueError:
-        raise ValueError("Size should be static for autoreparametrization.")
+    _, size, *_ = node.inputs
+    eval_size = size.eval()
```

Comment:
How is this better? You can do `constant_fold(raise_if_not_constant=False)`

Reply:
I remember hitting missing input errors, but I was doin plain eval

Reply:
Hmm the problem is you do need constant values with the shares variable

Reply:
ok, I reverted this due to this uncertainty. I only need this shape once, the computation of it can be addressed better. I saw in one place you did manual function compute with linker="py" and no optimizations, this would be suitable here as well but I do not want to duplicate code

Reply:
You can do `eval(mode="FAST_COMPILE")` if you want it to be fast. Still does some rewrites but not much and avoids C compilation